### PR TITLE
POL-675 - Optimize Azure Unattached Volumes

### PR DESCRIPTION
**Pull request by lawrethao**
### Description

This change updates the Azure Unused volumes policy to reduce the time and number of api calls to azure insights

### Issues Resolved

This fixes issues of timeouts for clients past the 2 hour mark

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/azure/unattached_volumes/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## v3.1
+
+- update policy for optimization, changes go as follows:
+  - removal of filtering on events and combining data of disks and activity
+  - iterate through disk rather than subscriptions to lower number of api requests due to pagination
```
Comment (taterdog590) [inline cost/azure/unattached_volumes/CHANGELOG.md:7]: iterate through filtered unattached disks

**Review by taterdog590 — APPROVED** [review]
LGTM

**Review by bryankaraffa — APPROVED** [review]

File: cost/azure/unattached_volumes/azure_delete_unattached_volumes.pt [hunk 2]
```diff
@@ -346,7 +342,7 @@ script "js_get_logactivity", type: "javascript" do
     var expectedDate=new Date();
     expectedDate.setDate(expectedDate.getDate()- parseInt(param_unattached_days));
     var date2=expectedDate.getFullYear()+'-'+(expectedDate.getMonth()+1)+'-'+expectedDate.getDate();
-    var filter= "eventTimestamp ge '"+date2+"T00:00:00Z' and eventTimestamp le '"+date1+"T23:59:59Z' and resourceProvider eq 'Microsoft.Compute' and resourceType eq 'Microsoft.Compute/VirtualMachines'"
+    var filter= "eventTimestamp ge '"+date2+"T00:00:00Z' and eventTimestamp le '"+date1+"T23:59:59Z' and eventChannels eq 'Admin, Operation' and resourceId eq '" + resourceID + "'"
```
Comment (taterdog590) [inline cost/azure/unattached_volumes/azure_delete_unattached_volumes.pt:345]: Levels should be included like:
and levels eq 'Critical,Error,Warning,Informational' at the very end of the filter query string.

Comment (taterdog590) [reply cost/azure/unattached_volumes/azure_delete_unattached_volumes.pt:345]: To use this, we would need to change line 355 to 'api-version': '2017-03-01-preview',

**Review by taterdog590 — APPROVED** [review]
LGTM

**Pull request merged**